# Nedbetaling av lån

**Læringsmål:**

* While-løkker med oppsamling

**Introduksjon:**

Annuitetslån er lån der man betaler et like stort beløp i det antall terminer som lånet løper (med mulig unntak av aller siste termin). Ved slike lån går mye av dette terminbeløpet til renter i starten, etter hvert blir en stadig større andel av terminbeløpet avdrag (tilbakebetaling). I tabellen under har vi vist hvordan et annuitetslån på $1\ 000 \text{ kr}$ nedbetales over $3$ terminer med en terminrente på $5 \text{ \%}$ og terminbeløp på $400 \text{ kr}$. Legg merke til at det siste terminbeløpet er redusert, da avdraget ikke skal overskride restgjelden.

| Termin | Terminbeløp | Rente | Avdrag | Restgjeld |
| --: | --: | --: | --: | --: |
| 0 | | | | 1000.0 |
| 1 | 400.0 | 50.0 | 350.0 | 650.0 |
| 2 | 400.0 | 32.5 | 367.5 | 282.5 |
| 3 | 296.6 | 14.1 | 282.5 | 0.0 |

Det som betales i rente i en termin regnes ut med formel 1:
$$\text{rente} = \text{restgjeld} \cdot \frac{\text{terminrente i prosent}}{100} \tag{1}$$
Avdragsbeløpet for hver termin vil da være gitt ved formelen : $\text{avdrag} = \text{terminbeløp} - \text{rente}$.

## a) 
Fullfør funksjonen `registrer_innbetaling` nedenfor. Funksjonen skal ta inn lånebeløp (gjeld), rentesats i prosent og terminbeløp som parametre, og returnere restgjelden som gjenstår etter innbetalingen.

In [ ]:
def registrer_innbetaling(gjeld, rentesats, terminbelop):
    # SKRIV DIN KODE HER!
    rente = gjeld * rentesats / 100
    avdrag = terminbelop - rente
    
    if avdrag > gjeld:
        avdrag = gjeld

    restgjeld = gjeld - avdrag

    # Beregn restgjeld gitt gjeld, rentesats og terminbeløp                 # ? Hehhh, hvorfor var dette allerede skrevet inn i oppgaven?
    # restgjeld = gjeld - (rentesats * terminbelop / 100)                   # ????????????
    
    return restgjeld

# Skriv funksjonskall for å sjekke funksjonen nedenfor:
print(f"{registrer_innbetaling(1000, 5, 400) = }")
print(f"{registrer_innbetaling(650, 5, 400) = }")



registrer_innbetaling(1000, 5, 400) = 650.0
registrer_innbetaling(650, 5, 400) = 282.5


Tips: du kan bruke funksjonskall med parametrene fra eksempelet ovenfor til å sjekke om funksjonen din fungerer slik den skal:
```Python
print(f"{registrer_innbetaling(1000, 5, 400) = }")
print(f"{registrer_innbetaling(650, 5, 400) = }")
```
Funksjonskallene skal gi følgende utskrift:<br>
`registrer_innbetaling(1000, 5, 400) = 650.0`<br>
`registrer_innbetaling(650, 5, 400) = 282.5`

## b) 
Bruk en `while`-løkke til å fullføre funksjonen `finn_nedbetalingstid`, som bruker gjentatte kall til `registrer_innbetaling` frem til hele lånet er nedbetalt. Funksjonen skal returnere nedbetalingstiden gitt i antall terminer (hvor mange ganger må terminbeløpet betales inn for at lånet skal være nedbetalt).



In [10]:
def finn_nedbetalingstid(gjeld, rentesats, terminbelop):
    # SKRIV DIN KODE HER!
    antall_terminer = 0
    restgjeld = gjeld
    while restgjeld > 0:
        restgjeld = registrer_innbetaling(restgjeld, rentesats, terminbelop)
        antall_terminer += 1
    return antall_terminer

# Skriv funksjonskall for å sjekke funksjonen nedenfor:
print(f"{finn_nedbetalingstid(1000, 5, 400) = }")

finn_nedbetalingstid(1000, 5, 400) = 3


Tips: du kan bruke et funksjonskall med parametrene fra eksempelet i starten av oppgaven til å sjekke om funksjonen din fungerer slik den skal:
```Python
print(f"{finn_nedbetalingstid(1000, 5, 400) = }")
```
Skal gi følgende utskrift:<br>
`finn_nedbetalingstid(1000, 5, 400) = 3`

## c)
Fullfør funksjonen `finn_samlet_rente` som returnerer det totale rentebeløpet som er innbetalt over hele låneperioden.

In [12]:
def finn_samlet_rente(gjeld, rentesats, terminbelop):
    # SKRIV DIN KODE HER!
    samlet_rente = 0
    restgjeld = gjeld
    while restgjeld > 0:
        rente = restgjeld * rentesats / 100
        samlet_rente += rente
        avdrag = terminbelop - rente
        if avdrag > restgjeld:
            avdrag = restgjeld
        restgjeld -= avdrag
    return samlet_rente

# Skriv funksjonskall for å sjekke funksjonen nedenfor:
print(f"{finn_samlet_rente(1000, 5, 400) = }")

finn_samlet_rente(1000, 5, 400) = 96.625


Tips: du kan bruke et funksjonskall med parametrene fra eksempelet i starten av oppgaven til å sjekke om funksjonen din fungerer slik den skal. Eksempel:
```Python
print(f"{finn_samlet_rente(1000, 5, 400) = }")
```
Skal gi følgende utskrift:<br>
`finn_samlet_rente(1000, 5, 400) = 96.625`